dashboard/app/controllers/experiments_controller.rb

```diff
@@ -22,13 +22,19 @@ def set_course_experiment
     redirect_to '/', flash: {notice: "You have successfully joined the experiment '#{params[:experiment_name]}'."}
   end
 
-  VALID_EXPERIMENTS = ['2018-teacher-experience'] + Experiment::PILOT_EXPERIMENTS.collect {|p| p[:name] if p[:allow_joining_via_url]}.compact
+  # Returns whether the given experiment can be joined (or left) via url.
+  def can_join_via_url?(experiment_name)
```

Comment:
For my understanding, could you explain the relationship between experiments and pilots?

Reply:
Maybe move to bottom, I think that's usually where we put helpers in controllers.

Reply:
Good question.  We have an experiments system which manages which users/sections/etc are assigned to experiments (identified by a string).  Script pilots are implemented as experiments using the pilot name as the experiment identifier.  One way to think about this is that script pilots are a particular type of experiment.

One interesting quirk of the current experiments system on the back-end is that there is no list of experiments in the database.  (The current `experiments` table would probably more accurately be called `experiment_assignments` or `experiment_participants`.)   I considered making the `pilots` table that i just added the list of experiments but wasn't sure how similar/different other experiments might be compared to pilots so I put pilots in a separate table for now.  It's possible that we will rename the table and unify the concept of experiments and pilots in the future.

Reply:
Good suggestion to move the function, will do that in a follow-up PR.